src/audio/dai-zephyr.c

```diff
@@ -1286,8 +1286,9 @@ int dai_zephyr_multi_endpoint_copy(struct dai_data **dd, struct comp_dev *dev,
 
 	/* return if nothing to copy */
 	if (!frames) {
+#ifndef CONFIG_ZEPHYR_DP_SCHEDULER
 		comp_warn(dev, "dai_zephyr_multi_endpoint_copy(): nothing to copy");
-
+#endif
```

Comment:
Maybe a fix for this 2021 bug?
- #4672 

Good things come to those who wait

Reply:
Oh, it's not that big. :)  It's like 500 line or so.  You should see some of the stuff I've inflicted on Zephyr.   I'm just wordy and like writing lots of commit messages.

I'll split out what I can, but really this is just a big rework of AEC.

Reply:
I did notice the low lines/commits ratio and I think that is great.

>  but really this is just a big rework of AEC.

I noticed that too but I also noticed some small fixes that really don't seem to have any connection (besides you founding and fixing them while working on AEC). These are almost "invisible" here: among other issues, many potential reviewers will go "AEC? Not my job." It wouldn't be a big deal if they were all purely cosmetic but that's not the case.

> I'll split out what I can,

Thank you!

Reply:
Commit moved to #8621